README.md
# Docker-Irisnet

Dockerized Iris Node

---

[![MicroBadger Version](https://images.microbadger.com/badges/version/ryanhendricks/docker-irisnet.svg)](https://microbadger.com/images/ryanhendricks/docker-irisnet)
[![CircleCI (all branches)](https://img.shields.io/circleci/project/github/RyanHendricks/docker-irisnet.svg?label=build&logo=circleci&logoColor=white)](https://circleci.com/gh/RyanHendricks/docker-irisnet)
[![Docker Pulls](https://img.shields.io/docker/pulls/ryanhendricks/docker-irisnet.svg?logo=docker&logoColor=white)](https://hub.docker.com/r/ryanhendricks/docker-irisnet)

[![MicroBadger Version](https://images.microbadger.com/badges/version/ryanhendricks/docker-irisnet:v0.15.3.svg)](https://microbadger.com/images/ryanhendricks/docker-irisnet:v0.15.3)
[![MicroBadger Layers (latest)](https://img.shields.io/microbadger/layers/ryanhendricks/docker-irisnet/latest.svg?logo=docker&logoColor=white)](https://microbadger.com/images/ryanhendricks/docker-irisnet)


## Getting Started

- Customize the configuration within scripts/config.sh
- Change ENV variables in Dockerfile (if needed)

## Start Node

```bash
docker-compose up -d --build
```

## Notes

This setup is meant to provide a full node only and not recommended for running a validator node. Although with some additional configuration to the node and the server where it will be running it is entirely possible to do so.


.github/CHANGELOG.md
# Changelog

## [v1.1.0](https://github.com/RyanHendricks/docker-irisnet/tree/v1.1.0) (2021-03-31)

[Full Changelog](https://github.com/RyanHendricks/docker-irisnet/compare/v1.0.1...v1.1.0)

## [v1.0.1](https://github.com/RyanHendricks/docker-irisnet/tree/v1.0.1) (2021-02-21)

[Full Changelog](https://github.com/RyanHendricks/docker-irisnet/compare/v0.16.2...v1.0.1)

## [v0.16.2](https://github.com/RyanHendricks/docker-irisnet/tree/v0.16.2) (2020-04-13)

[Full Changelog](https://github.com/RyanHendricks/docker-irisnet/compare/v0.16.0...v0.16.2)

## [v0.16.0](https://github.com/RyanHendricks/docker-irisnet/tree/v0.16.0) (2019-12-07)

[Full Changelog](https://github.com/RyanHendricks/docker-irisnet/compare/v0.15.3...v0.16.0)

## [v0.15.3](https://github.com/RyanHendricks/docker-irisnet/tree/v0.15.3) (2019-10-07)

[Full Changelog](https://github.com/RyanHendricks/docker-irisnet/compare/5f617cfbca5b128c3c22a05cb42c85398fef1fc3...v0.15.3)



\* *This Changelog was automatically generated by [github_changelog_generator](https://github.com/github-changelog-generator/github-changelog-generator)*


config/test.sh
#!/bin/sh


# exit script on any error
set -e


cat ./config.toml > test.toml


# cat > config.toml << EOF
# # This is a TOML config file.
# # For more information, see https://github.com/toml-lang/toml

# ##### main base config options #####

# # TCP or UNIX socket address of the ABCI application,
# # or the name of an ABCI application compiled in with the Tendermint binary
# proxy_app = "tcp://127.0.0.1:26658"

# # A custom human readable name for this node
# moniker = "${MONIKER:-iris_moniker}"

# # If this node is many blocks behind the tip of the chain, FastSync
# # allows them to catchup quickly by downloading blocks in parallel
# # and verifying their commits
# fast_sync = true

# # If the blockchain is deprecated, run node with Deprecated will
# # work in query only mode. Consensus engine and p2p gossip will be
# # shutdown
# deprecated = false

# # Database backend: leveldb | memdb | cleveldb
# db_backend = "leveldb"

# # Database directory
# db_dir = "data"

# # Output level for logging, including package level options
# log_level = "main:info,state:info,*:error"

# # Output format: 'plain' (colored text) or 'json'
# log_format = "plain"

# ##### additional base config options #####

# # Path to the JSON file containing the initial validator set and other meta data
# genesis_file = "config/genesis.json"

# # Path to the JSON file containing the private key to use as a validator in the consensus protocol
# priv_validator_file = "config/priv_validator.json"

# # TCP or UNIX socket address for Tendermint to listen on for
# # connections from an external PrivValidator process
# priv_validator_laddr = ""

# # Path to the JSON file containing the private key to use for node authentication in the p2p protocol
# node_key_file = "config/node_key.json"

# # Mechanism to connect to the ABCI application: socket | grpc
# abci = "socket"

# # TCP or UNIX socket address for the profiling server to listen on
# prof_laddr = "localhost:6060"

# # If true, query the ABCI app on connecting to a new peer
# # so the app can decide if we should keep the connection or not
# filter_peers = false

# ##### advanced configuration options #####

# ##### rpc server configuration options #####
# [rpc]

# # TCP or UNIX socket address for the RPC server to listen on
# laddr = "tcp://0.0.0.0:26657"

# # A list of origins a cross-domain request can be executed from
# # Default value '[]' disables cors support
# # Use '["*"]' to allow any origin
# cors_allowed_origins = ["*"]

# # A list of methods the client is allowed to use with cross-domain requests
# cors_allowed_methods = ["HEAD", "GET", "POST", ]

# # A list of non simple headers the client is allowed to use with cross-domain requests
# cors_allowed_headers = ["Origin", "Accept", "Content-Type", "X-Requested-With", "X-Server-Time", ]

# # TCP or UNIX socket address for the gRPC server to listen on
# # NOTE: This server only supports /broadcast_tx_commit
# grpc_laddr = ""

# # Maximum number of simultaneous connections.
# # Does not include RPC (HTTP&WebSocket) connections. See max_open_connections
# # If you want to accept a larger number than the default, make sure
# # you increase your OS limits.
# # 0 - unlimited.
# # Should be < {ulimit -Sn} - {MaxNumInboundPeers} - {MaxNumOutboundPeers} - {N of wal, db and other open files}
# # 1024 - 40 - 10 - 50 = 924 = ~900
# grpc_max_open_connections = 900

# # Activate unsafe RPC commands like /dial_seeds and /unsafe_flush_mempool
# unsafe = false

# # Maximum number of simultaneous connections (including WebSocket).
# # Does not include gRPC connections. See grpc_max_open_connections
# # If you want to accept a larger number than the default, make sure
# # you increase your OS limits.
# # 0 - unlimited.
# # Should be < {ulimit -Sn} - {MaxNumInboundPeers} - {MaxNumOutboundPeers} - {N of wal, db and other open files}
# # 1024 - 40 - 10 - 50 = 924 = ~900
# max_open_connections = 900

# ##### peer to peer configuration options #####
# [p2p]

# # Address to listen for incoming connections
# laddr = "tcp://0.0.0.0:26656"

# # Address to advertise to peers for them to dial
# # If empty, will use the same port as the laddr,
# # and will introspect on the listener or use UPnP
# # to figure out the address.
# external_address = ""

# # Comma separated list of seed nodes to connect to
# seeds = "6a6de770deaa4b8c061dffd82e9c7f4d40c2165d@seed-1.mainnet.irisnet.org:26656,a17d7923293203c64ba75723db4d5f28e642f469@seed-2.mainnet.irisnet.org:26656"

# # Comma separated list of nodes to keep persistent connections to
# persistent_peers = ""

# # UPNP port forwarding
# upnp = false

# # Path to address book
# addr_book_file = "config/addrbook.json"

# # Set true for strict address routability rules
# # Set false for private or local networks
# addr_book_strict = false

# # Maximum number of inbound peers
# max_num_inbound_peers = 40

# # Maximum number of outbound peers to connect to, excluding persistent peers
# max_num_outbound_peers = 10

# # Time to wait before flushing messages out on the connection
# flush_throttle_timeout = "100ms"

# # Maximum size of a message packet payload, in bytes
# max_packet_msg_payload_size = 2048

# # Rate at which packets can be sent, in bytes/second
# send_rate = 10240000

# # Rate at which packets can be received, in bytes/second
# recv_rate = 10240000

# # Set true to enable the peer-exchange reactor
# pex = true

# # Seed mode, in which node constantly crawls the network and looks for
# # peers. If another node asks it for addresses, it responds and disconnects.
# #
# # Does not work if the peer-exchange reactor is disabled.
# seed_mode = false

# # Comma separated list of peer IDs to keep private (will not be gossiped to other peers)
# private_peer_ids = ""

# # Toggle to disable guard against peers connecting from the same ip.
# allow_duplicate_ip = false

# # Peer connection configuration.
# handshake_timeout = "20s"
# dial_timeout = "3s"

# ##### mempool configuration options #####
# [mempool]

# recheck = true
# broadcast = true
# wal_dir = ""

# # size of the mempool
# size = 5000

# # size of the cache (used to filter transactions we saw earlier)
# cache_size = 10000

# ##### consensus configuration options #####
# [consensus]

# wal_file = "data/cs.wal/wal"

# timeout_propose = "3s"
# timeout_propose_delta = "500ms"
# timeout_prevote = "1s"
# timeout_prevote_delta = "500ms"
# timeout_precommit = "1s"
# timeout_precommit_delta = "500ms"
# timeout_commit = "5s"

# # Make progress as soon as we have all the precommits (as if TimeoutCommit = 0)
# skip_timeout_commit = false

# # EmptyBlocks mode and possible interval between empty blocks
# create_empty_blocks = true
# create_empty_blocks_interval = "0s"

# # Reactor sleep duration parameters
# peer_gossip_sleep_duration = "100ms"
# peer_query_maj23_sleep_duration = "2s"

# # Block time parameters. Corresponds to the minimum time increment between consecutive blocks.
# blocktime_iota = "1s"

# ##### transactions indexer configuration options #####
# [tx_index]

# # What indexer to use for transactions
# #
# # Options:
# #   1) "null"
# #   2) "kv" (default) - the simplest possible indexer, backed by key-value storage (defaults to levelDB; see DBBackend).
# indexer = "kv"

# # Comma-separated list of tags to index (by default the only tag is "tx.hash")
# #
# # You can also index transactions by height by adding "tx.height" tag here.
# #
# # It's recommended to index only a subset of tags due to possible memory
# # bloat. This i#!/bin/sh


# # exit script on any error
# set -es, of course, depends on the indexer's DB and the volume of
# # transactions.
# index_tags = "action,tx.height"

# # When set to true, tells indexer to index all tags (predefined tags:
# # "tx.hash", "tx.height" and all tags from DeliverTx responses).
# #
# # Note this may be not desirable (see the comment above). IndexTags has a
# # precedence over IndexAllTags (i.e. when given both, IndexTags will be
# # indexed).
# index_all_tags = false

# ##### instrumentation configuration options #####
# [instrumentation]

# # When true, Prometheus metrics are served under /metrics on
# # PrometheusListenAddr.
# # Check out the documentation for the list of available metrics.
# prometheus = true

# # Address to listen for Prometheus collector(s) connections
# prometheus_listen_addr = ":26660"

# # Maximum number of simultaneous connections.
# # If you want to accept a larger number than the default, make sure
# # you increase your OS limits.
# # 0 - unlimited.
# max_open_connections = 0

# # Instrumentation namespace
# namespace = "tendermint"

# EOF


scripts/config.sh
#!/bin/sh

# mkdir /.iris
MONIKER=${MONIKER:-iris_moniker}
iris init --moniker=$MONIKER --home=$IRISHOME --chain-id=$CHAIN_ID
cd $IRISHOME/config/
rm genesis.json
rm config.toml
wget ${GENESIS_URL:-https://raw.githubusercontent.com/irisnet/betanet/master/config/genesis.json}


cat > config.toml << EOF
# This is a TOML config file.
# For more information, see https://github.com/toml-lang/toml

##### main base config options #####

# TCP or UNIX socket address of the ABCI application,
# or the name of an ABCI application compiled in with the Tendermint binary
proxy_app = "tcp://127.0.0.1:26658"

# A custom human readable name for this node
moniker = "${MONIKER}"

# If this node is many blocks behind the tip of the chain, FastSync
# allows them to catchup quickly by downloading blocks in parallel
# and verifying their commits
fast_sync = true

# If the blockchain is deprecated, run node with Deprecated will
# work in query only mode. Consensus engine and p2p gossip will be
# shutdown
deprecated = false

# Database backend: leveldb | memdb | cleveldb
db_backend = "leveldb"

# Database directory
db_dir = "data"

# Output level for logging, including package level options
log_level = "main:info,state:info,*:error"

# Output format: 'plain' (colored text) or 'json'
log_format = "plain"

##### additional base config options #####

# Path to the JSON file containing the initial validator set and other meta data
genesis_file = "config/genesis.json"

# Path to the JSON file containing the private key to use as a validator in the consensus protocol
priv_validator_file = "config/priv_validator.json"

# TCP or UNIX socket address for Tendermint to listen on for
# connections from an external PrivValidator process
priv_validator_laddr = ""

# Path to the JSON file containing the private key to use for node authentication in the p2p protocol
node_key_file = "config/node_key.json"

# Mechanism to connect to the ABCI application: socket | grpc
abci = "socket"

# TCP or UNIX socket address for the profiling server to listen on
prof_laddr = "localhost:6060"

# If true, query the ABCI app on connecting to a new peer
# so the app can decide if we should keep the connection or not
filter_peers = false

##### advanced configuration options #####

##### rpc server configuration options #####
[rpc]

# TCP or UNIX socket address for the RPC server to listen on
laddr = "tcp://0.0.0.0:26657"

# A list of origins a cross-domain request can be executed from
# Default value '[]' disables cors support
# Use '["*"]' to allow any origin
cors_allowed_origins = ["*"]

# A list of methods the client is allowed to use with cross-domain requests
cors_allowed_methods = ["HEAD", "GET", "POST", ]

# A list of non simple headers the client is allowed to use with cross-domain requests
cors_allowed_headers = ["Origin", "Accept", "Content-Type", "X-Requested-With", "X-Server-Time", ]

# TCP or UNIX socket address for the gRPC server to listen on
# NOTE: This server only supports /broadcast_tx_commit
grpc_laddr = ""

# Maximum number of simultaneous connections.
# Does not include RPC (HTTP&WebSocket) connections. See max_open_connections
# If you want to accept a larger number than the default, make sure
# you increase your OS limits.
# 0 - unlimited.
# Should be < {ulimit -Sn} - {MaxNumInboundPeers} - {MaxNumOutboundPeers} - {N of wal, db and other open files}
# 1024 - 40 - 10 - 50 = 924 = ~900
grpc_max_open_connections = 900

# Activate unsafe RPC commands like /dial_seeds and /unsafe_flush_mempool
unsafe = false

# Maximum number of simultaneous connections (including WebSocket).
# Does not include gRPC connections. See grpc_max_open_connections
# If you want to accept a larger number than the default, make sure
# you increase your OS limits.
# 0 - unlimited.
# Should be < {ulimit -Sn} - {MaxNumInboundPeers} - {MaxNumOutboundPeers} - {N of wal, db and other open files}
# 1024 - 40 - 10 - 50 = 924 = ~900
max_open_connections = 900

##### peer to peer configuration options #####
[p2p]

# Address to listen for incoming connections
laddr = "tcp://0.0.0.0:26656"

# Address to advertise to peers for them to dial
# If empty, will use the same port as the laddr,
# and will introspect on the listener or use UPnP
# to figure out the address.
external_address = ""

# Comma separated list of seed nodes to connect to
seeds = "6a6de770deaa4b8c061dffd82e9c7f4d40c2165d@seed-1.mainnet.irisnet.org:26656,a17d7923293203c64ba75723db4d5f28e642f469@seed-2.mainnet.irisnet.org:26656"

# Comma separated list of nodes to keep persistent connections to
persistent_peers = ""

# UPNP port forwarding
upnp = false

# Path to address book
addr_book_file = "config/addrbook.json"

# Set true for strict address routability rules
# Set false for private or local networks
addr_book_strict = false

# Maximum number of inbound peers
max_num_inbound_peers = 40

# Maximum number of outbound peers to connect to, excluding persistent peers
max_num_outbound_peers = 10

# Time to wait before flushing messages out on the connection
flush_throttle_timeout = "100ms"

# Maximum size of a message packet payload, in bytes
max_packet_msg_payload_size = 2048

# Rate at which packets can be sent, in bytes/second
send_rate = 10240000

# Rate at which packets can be received, in bytes/second
recv_rate = 10240000

# Set true to enable the peer-exchange reactor
pex = true

# Seed mode, in which node constantly crawls the network and looks for
# peers. If another node asks it for addresses, it responds and disconnects.
#
# Does not work if the peer-exchange reactor is disabled.
seed_mode = false

# Comma separated list of peer IDs to keep private (will not be gossiped to other peers)
private_peer_ids = ""

# Toggle to disable guard against peers connecting from the same ip.
allow_duplicate_ip = false

# Peer connection configuration.
handshake_timeout = "20s"
dial_timeout = "3s"

##### mempool configuration options #####
[mempool]

recheck = true
broadcast = true
wal_dir = ""

# size of the mempool
size = 5000

# size of the cache (used to filter transactions we saw earlier)
cache_size = 10000

##### consensus configuration options #####
[consensus]

wal_file = "data/cs.wal/wal"

timeout_propose = "3s"
timeout_propose_delta = "500ms"
timeout_prevote = "1s"
timeout_prevote_delta = "500ms"
timeout_precommit = "1s"
timeout_precommit_delta = "500ms"
timeout_commit = "5s"

# Make progress as soon as we have all the precommits (as if TimeoutCommit = 0)
skip_timeout_commit = false

# EmptyBlocks mode and possible interval between empty blocks
create_empty_blocks = true
create_empty_blocks_interval = "0s"

# Reactor sleep duration parameters
peer_gossip_sleep_duration = "100ms"
peer_query_maj23_sleep_duration = "2s"

# Block time parameters. Corresponds to the minimum time increment between consecutive blocks.
blocktime_iota = "1s"

##### transactions indexer configuration options #####
[tx_index]

# What indexer to use for transactions
#
# Options:
#   1) "null"
#   2) "kv" (default) - the simplest possible indexer, backed by key-value storage (defaults to levelDB; see DBBackend).
indexer = "kv"

# Comma-separated list of tags to index (by default the only tag is "tx.hash")
#
# You can also index transactions by height by adding "tx.height" tag here.
#
# It's recommended to index only a subset of tags due to possible memory
# bloat. This is, of course, depends on the indexer's DB and the volume of
# transactions.
index_tags = "action,tx.height"

# When set to true, tells indexer to index all tags (predefined tags:
# "tx.hash", "tx.height" and all tags from DeliverTx responses).
#
# Note this may be not desirable (see the comment above). IndexTags has a
# precedence over IndexAllTags (i.e. when given both, IndexTags will be
# indexed).
index_all_tags = false

##### instrumentation configuration options #####
[instrumentation]

# When true, Prometheus metrics are served under /metrics on
# PrometheusListenAddr.
# Check out the documentation for the list of available metrics.
prometheus = true

# Address to listen for Prometheus collector(s) connections
prometheus_listen_addr = ":26660"

# Maximum number of simultaneous connections.
# If you want to accept a larger number than the default, make sure
# you increase your OS limits.
# 0 - unlimited.
max_open_connections = 0

# Instrumentation namespace
namespace = "tendermint"

EOF
# config.toml $IRISHOME/config/config.toml
